src/uhs/transaction/validation.cpp

```diff
@@ -403,4 +403,24 @@ namespace cbdc::transaction::validation {
                          err);
         return str;
     }
+
+    auto check_attestations(
+        const transaction::compact_tx& tx,
+        const std::unordered_set<pubkey_t, hashing::null>& pubkeys,
+        size_t threshold) -> bool {
+        if(tx.m_attestations.size() < threshold) {
+            return false;
+        }
+
+        if(!std::all_of(tx.m_attestations.begin(),
+                        tx.m_attestations.end(),
+                        [&](const auto& att) {
+                            return pubkeys.find(att.first) != pubkeys.end()
+                                && tx.verify(secp_context.get(), att);
+                        })) {
+            return false;
+        }
+
+        return true;
```

Comment:
```suggestion
        return std::all_of(
            tx.m_attestations.begin(),
            tx.m_attestations.end(),
            [&](const auto& att) {
                return pubkeys.find(att.first) != pubkeys.end()
                       && tx.verify(secp_context.get(), att);
            });
```

Any reason to avoid the immediate return?